api.go

```diff
@@ -397,15 +397,21 @@ func (c Client) executeMethod(method string, metadata requestMetadata) (res *htt
 		res, err = c.do(req)
 		if err != nil {
 			// Fail quickly here, no need to retry.
-			return nil, err
+			return res, err
 		}
```

Comment:
> when ever err is non-nil, res is undefined. Why would you return res instead of nil?
> Also local network errors, server disconnects needs retry as well.

Network errors will do separately, it is not part of this patch.  Be 'res' nil or non nil we shouldn't define the value. I am returning everything that is sent by the lower level call to .Do()